packages/babel-plugin-transform-modules-commonjs/README.md

```diff
@@ -130,3 +130,37 @@ function _interopRequireDefault(obj) {
 In cases where the auto-unwrapping of `default` is not needed, you can set the
 `noInterop` option to `true` to avoid the usage of the `interopRequireDefault`
 helper (shown in inline form above).
+
+### `lazy`
+
+`boolean`, `Array<string>`, or `(string) => boolean`, defaults to `false`
+
+Changes Babel's compiled `import` statements to be lazily evaluated when their
+imported bindings are used for the first time.
+
+This can improve initial load time of your module because evaluating
+dependencies up front is sometimes entirely un-necessary. This is especially
+the case when implementing a library module.
+
+The value of `lazy` has a few possible effects:
+
+* `false` - No lazy initialization of any imported module.
+* `true` - Do not lazy-initialize local `./foo` imports, but lazy-init `foo` dependencies.
+
+  Local paths are much more likely to have circular dependencies, which may break if loaded lazily,
+  so they are not lazy by default, whereas dependencies between independent modules are rarely cyclical.
+
+* `Array<string>` - Lazy-initialize all imports with source matching one of the given strings.
```

Comment:
do we need `Array<string>`, do you think it is a common case? I guess `boolean | (string) => boolean` would capture these case too. I was also thinking about more cases like "enable really everything" or "enable everything with a blacklist", but that can all be handled with the callback.

Reply:
The disadvantage of the callback is that it forces the user to use `.babelrc.js` instead of `.babelrc`

Reply:
Yeah the `.babelrc`/`package.json#babel` case was the reason I added the array. Seems like a whitelist was easy enough to support that I'd just throw it in.